hack/tests/e2e-ansible.sh

```diff
@@ -145,12 +145,17 @@ cat "$ROOTDIR/test/ansible-memcached/watches-foo-kind.yaml" >> memcached-operato
 pushd memcached-operator
 # Add a second Kind to test watching multiple GVKs
 operator-sdk add crd --kind=Foo --api-version=ansible.example.com/v1alpha1
-sed -i 's|\(FROM quay.io/operator-framework/ansible-operator\)\(:.*\)\?|\1:dev|g' build/Dockerfile
+sed -i".bak" -E -e 's/(FROM quay.io\/operator-framework\/ansible-operator)(:.*)?/\1:dev/g' build/Dockerfile; rm -f build/Dockerfile.bak
 operator-sdk build "$DEST_IMAGE"
 # If using a kind cluster, load the image into all nodes.
 load_image_if_kind "$DEST_IMAGE"
-sed -i "s|{{ REPLACE_IMAGE }}|$DEST_IMAGE|g" deploy/operator.yaml
-sed -i 's|{{ pull_policy.default..Always.. }}|Never|g' deploy/operator.yaml
+if [[ "$OSTYPE" == "darwin"* ]]; then
+  sed -i  " " "s|{{ REPLACE_IMAGE }}|$DEST_IMAGE|g" deploy/operator.yaml
```

Comment:
@joelanford just here that I could not find a solution because of "{{" which should be a string and is not.

Reply:
```sh
DEST_IMAGE_ESCAPED="$(printf '%s' "$DEST_IMAGE" | sed 's/[.[\/*^$]/\\&/g')"
sed -i".bak" -E -e "s/\{\{ REPLACE_IMAGE \}\}/$DEST_IMAGE_ESCAPED/g" deploy/operator.yaml; rm -f deploy/operator.yaml.bak
sed -i".bak" -E -e 's/\{\{ pull_policy.default..Always.. \}\}/Never/g' deploy/operator.yaml; rm -f deploy/operator.yaml.bak
```

Reply:
I think I was wrong originally when I said that `|` characters don't work with BSD sed. It seems that they do, in fact, work with both GNU sed and BSD sed. So I think we should use `|` everywhere we were using it before so that we don't have to escape the forward slashes in the regex patterns and replacements.